### PR TITLE
[ASTS] Targeted Sweep Part 7: Deleting Sweep Queue

### PR DESCRIPTION
**Pull request by mdaudali**
## General
**Before this PR**:
SweepQueue exists
**After this PR**:

SweepQueue no longer exists

==COMMIT_MSG==
==COMMIT_MSG==

**Priority**: P2

**Concerns / possible downsides (what feedback would you like?)**:
I don't really like exposing the components and metrics for testing. I can add the metrics to the components, but that doesn't entirely solve the problem. I'd considered some form of Partial builder style thing, but I wasn't sure it was worth the time.

The SettableFuture is also a bit annoying for tests. One option is to have TargetedSweeper expose the writer as a method, doing the settable future stuff internally - but that kinda goes against the whole inject rather than eject thing for non-test code.

I also don't particularly like making the stores public to get access to the TableRef, but it feels better than getting the table ref directly from TargetedSweepTableFactory
**Is documentation needed?**:

## Compatibility
**Does this PR create any API breaks (e.g. at the Java or HTTP layers) - if so, do we have compatibility?**:
No
**Does this PR change the persisted format of any data - if so, do we have forward and backward compatibility?**:
No
**The code in this PR may be part of a blue-green deploy. Can upgrades from previous versions safely coexist? (Consider restarts of blue or green nodes.)**:
Yes
**Does this PR rely on statements being true about other products at a deployment - if so, do we have correct product dependencies on these products (or other ways of verifying that these statements are true)?**:
No
**Does this PR need a schema migration?**
No
## Testing and Correctness
**What, if any, assumptions are made about the current state of the world? If they change over time, how will we find out?**:
None
**What was existing testing like? What have you done to improve it?**:
Tests updated
**If this PR contains complex concurrent or asynchronous code, is it correct? The onus is on the PR writer to demonstrate this.**:
N/A
**If this PR involves acquiring locks or other shared resources, how do we ensure that these are always released?**:
N/A
## Execution
**How would I tell this PR works in production? (Metrics, logs, etc.)**:
No change
**Has the safety of all log arguments been decided correctly?**:
No change
**Will this change significantly affect our spending on metrics or logs?**:
No change
**How would I tell that this PR does not work in production? (monitors, etc.)**:
No change
**If this PR does not work as expected, how do I fix that state? Would rollback be straightforward?**:
Rollback
**If the above plan is more complex than “recall and rollback”, please tag the support PoC here (if it is the end of the week, tag both the current and next PoC)**:
N/A
## Scale
No change

## Development Process
**Where should we start reviewing?**:
TransactionManagers, TargetedSweeper, MultiWriter thing

**If this PR is in excess of 500 lines excluding versions lock-files, why does it not make sense to split it?**:
The main changes are to tests, deleting SweepQueue, and changing around TargetedSweeper.
I could split it, but it felt like it made more sense in this one PR to switch over

**Please tag any other people who should be aware of this PR**:
@jeremyk-91
@raiju

### CODE REVIEW

File: atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/TargetedSweeper.java [hunk 1]
```diff
@@ -208,54 +235,37 @@ public void runInBackground() {
...
+    public SweepQueueComponents components() {
+        assertInitialized();
+        return components;
+    }
+
+    // @VisibleForTesting
+    public TargetedSweepMetrics metrics() {
+        assertInitialized();
+        return metrics;
+    }
```
Comment (jeremyk-91) [inline atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/TargetedSweeper.java:267]: check: are these intentional? My guess is yes

File: atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/MultiTableSweepQueueWriter.java [hunk 2]
```diff
@@ -41,30 +39,13 @@ default void enqueue(Map<TableReference, ? extends Map<Cell, byte[]>> writes, lo
...
 
-    /**
-     * This method must be implemented if asynchronous initialization is necessary for the implementation. This is
-     * generally the case if the transaction manager allows asynchronous initialization since there is no guarantee
-     * the underlying kvs is ready at object creation time.
-     *
-     * @param txManager the transaction manager performing the callback
-     */
-    @Override
-    default void initialize(TransactionManager txManager) {
```
Comment (jeremyk-91) [inline atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/MultiTableSweepQueueWriter.java:52]: checked in large internal product that asynchronous initialization is NOT necessary for the implementation there (and in our usage, the way in which TM initialisation works means that it is also not necessary for us)

File: atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/TargetedSweeper.java (hunk 1 above)
Comment (jeremyk-91) [reply atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/TargetedSweeper.java:267]: discussed offline: yes these are used by tests that aren't in the same package

**Review by jeremyk-91 — APPROVED** [review]
👍

**Comment by mdaudali** [comment]
👍

**Pull request merged**